assets/js/modules/analytics-4/datastore/audience-settings.js

```diff
@@ -0,0 +1,278 @@
+} );
+
+const fetchSaveAudienceSettingsStore = createFetchStore( {
+	baseName: 'saveAudienceSettings',
+	controlCallback: ( settings ) =>
+		API.set( 'modules', 'analytics-4', 'save-audience-settings', {
+			settings,
+		} ),
+	reducerCallback: fetchStoreReducerCallback,
+	argsToParams: ( settings ) => settings,
+	validateParams: validateAudienceSettings,
+} );
+
+// Actions
+const SET_AVAILABLE_AUDIENCES = 'SET_AVAILABLE_AUDIENCES';
+const SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY =
+	'SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY';
+
+const baseInitialState = {
+	audienceSettings: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Sets the available audiences.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Array} availableAudiences Available audiences resource names.
+	 * @return {Object} Redux-style action.
+	 */
+	setAvailableAudiences( availableAudiences ) {
+		invariant(
+			Array.isArray( availableAudiences ),
+			'Available audiences should be an array.'
+		);
+
+		return {
+			type: SET_AVAILABLE_AUDIENCES,
+			payload: { availableAudiences },
+		};
+	},
+
+	/**
+	 * Sets the audience segmentation setup completed by.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {number} audienceSegmentationSetupCompletedBy Audience segmentation setup completed by.
+	 * @return {Object} Redux-style action.
+	 */
+	setAudienceSegmentationSetupCompletedBy(
+		audienceSegmentationSetupCompletedBy
+	) {
+		// Should be integer.
+		invariant(
+			typeof audienceSegmentationSetupCompletedBy === 'number',
+			'Audience segmentation setup completed by should be an integer.'
+		);
+
+		return {
+			type: SET_AUDIENCE_SEGMENTATION_SETUP_COMPLETED_BY,
+			payload: { audienceSegmentationSetupCompletedBy },
+		};
+	},
+
+	/**
+	 * Saves the audience settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} settings Audience settings to save.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	saveAudienceSettings: createValidatedAction(
+		( settings ) => {
+			validateAudienceSettings( settings );
+		},
+		function* ( settings ) {
+			yield clearError( 'saveAudienceSettings', [] );
+
+			const { response, error } =
+				yield fetchSaveAudienceSettingsStore.actions.fetchSaveAudienceSettings(
+					settings
+				);
+
+			if ( error ) {
+				yield receiveError( error, 'saveUserAudienceSettings', [] );
+			}
+
+			return { response, error };
+		}
+	),
+};
+
+const baseResolvers = {
+	*getAvailableAudiences() {
+		const registry = yield commonActions.getRegistry();
+		const { select, dispatch } = registry;
+
+		const audiences = select( MODULES_ANALYTICS_4 ).getAvailableAudiences();
+
+		if ( audiences === null ) {
+			const { response, error } = yield commonActions.await(
+				dispatch( MODULES_ANALYTICS_4 ).syncAvailableAudiences()
+			);
+
+			if ( error ) {
+				return { error };
+			}
+
+			dispatch( MODULES_ANALYTICS_4 ).setAvailableAudiences( response );
+		}
+	},
+};
+
+const baseReducer = ( state, { type, payload } ) => {
```

Comment:
We might as well use `createReducer()` here and simplify the reducer, seeing as it's used earlier in the file.

```suggestion
const baseReducer = createReducer( ( state, { type, payload } ) => {
```

Then update the the Immer-enabled reducer logic to directly mutate `state`.